.github/FORK_PREVIEW_SETUP.md

```diff
@@ -0,0 +1,178 @@
+
+### Content Validation
+
+- Checks for executable files in content directories
+- Validates blog post frontmatter structure
+- Detects potentially unsafe content patterns
+- Warns about oversized images (>2MB)
+- Runs textlint and prettier if available
+
+### Access Controls
+
+- Only processes PRs from forked repositories
+- Uses `pull_request_target` with explicit fork checkout
+- Separates untrusted code execution from credential access
+
+## Workflow Behavior
+
+### Build Process
+
+1. Detects build system (npm/yarn/pnpm/hugo/custom)
+2. Runs content validation and security checks
+3. Installs dependencies and runs linting
+4. Builds the site
+5. Packages output as artifact
+
+### Deployment Process
+
+1. Downloads build artifact from Stage 1
+2. Deploys to Cloudflare Pages using API
+3. Creates stable preview URL: `https://{hash}.ddev-com-fork-previews.pages.dev`
+4. Comments preview URL on the PR
+5. Updates comment on subsequent pushes
+
+### PR Lifecycle
+
+- **Opened/Updated**: Creates or updates preview
+- **Closed**: Adds closure note (preview remains accessible)
+- **Draft**: Still builds and deploys (no special handling)
+
+## Troubleshooting
+
+### Build Failures
+
+- Check build logs in GitHub Actions
+- Ensure dependencies install correctly
+- Verify build command produces output directory
+
+### Missing Secrets
+
+- Workflow will fail with clear error messages
+- Verify all three secrets are set correctly
+- Check Cloudflare API token permissions
+
+### Content Validation Errors
+
+- Review security check output
+- Fix frontmatter issues in blog posts
+- Address linting warnings locally with:
+  - `ddev npm run textlint:fix`
+  - `ddev npm run prettier:fix`
+
+### Preview URL Issues
+
+- Verify `ddev-com-fork-previews` Cloudflare Pages project exists and is accessible
+- Check account ID matches the project's organization
+- Ensure `CF_PAGES_PROJECT` is set to `ddev-com-fork-previews`
+
+## Manual Testing
+
+To test the workflow:
+
+1. Create a test fork of the repository
+2. Make a content change (e.g., add a blog post)
+3. Open a PR from the fork
+4. Watch GitHub Actions for build/deploy progress
+5. Check for preview URL comment on the PR
+
+## Maintenance
+
+### Regular Tasks
+
+- Monitor Cloudflare Pages usage and costs
+- Review security warnings in build logs
+- Update dependencies in fork validation steps
+- Clean up old preview deployments if needed
+
+### Updates
+
+- Keep `cloudflare/pages-action` version current
+- Monitor Cloudflare API changes
+- Update content validation rules as needed
+
+## Known Issues & Future Improvements
+
+### URL Stability Problem
+
+Currently, Cloudflare's Direct Upload API creates a new deployment hash with each push, resulting in changing preview URLs like:
+
+- First push: `https://2e1a74f4.ddev-com-fork-previews.pages.dev`
+- Second push: `https://e98fa6ff.ddev-com-fork-previews.pages.dev`
+
+This breaks the ability to bookmark or share stable preview URLs for review purposes.
```

Comment:
There is stable URL available:

<img width="1248" height="341" alt="image" src="https://github.com/user-attachments/assets/d2438589-da7b-4b1d-a7a8-99b8c689188c" />

<img width="965" height="300" alt="image" src="https://github.com/user-attachments/assets/978dae6c-d90c-48ba-967f-d9c81a268d06" />